### DISCUSSION TITLE
How to read a private tag byte-by-byte

### DISCUSSION THREAD

**Discussion by bec1442** [fo-dicom/fo-dicom#1873] [Q&A]
I am trying to read private tag byte-by-byte so using this code

```
DicomTag privateTag = new DicomTag(0x3253, 0x1000);
 // Check if the private creator exists
 if (dicomDataset.Contains(privateTag))
 {
     Console.WriteLine($"Private tag {privateTag} exist in the dataset.");

     //// Get the value of the private tag
      byte[] privateTagData = dicomDataset.GetValues<byte>(privateTag);
     Console.WriteLine($"content {privateTagData[0]} ");
 }
 else
 {
     Console.WriteLine($"Private tag {privateTag} does not exist in the dataset.");
     
    Console.WriteLine($"These are the tags available in the dataset.");
     foreach (var ds in dicomDataset)
     {
         Console.WriteLine(ds);
     }
 }
```

But I get **"Private tag (3253,1000)does not exist in dataset is"** output message . 
 I also added the **foreach()** to see if the tag actually exist in the dataset and it does!
What am I doing wrong?  Any help is greatly appreciated.

**Comment by mrbean-bremen** [comment]
See the related issue - maybe your dataset does not have a valid Private Creator for the tag? Though I have never seen private tags without a private creator, it may have been created by some faulty application.

**Reply by bec1442** [reply]
The Dataset doe have a creator tag (3253,001O) and that one can be accessed.

**Comment by mrbean-bremen** [comment]
Copying my comment from the issue here:

A private tag is characterized by the group, the (1-byte) element offset and the Private Creator string, so to create a valid private tag, you need something like:
```cs
var privateTag = new DicomTag(0x3253, 0x00, "YOUR PRIVATE CREATOR");
// Check if private tag exist
if (dicomDataset.Contains(privateTag))
{
...
```
The upper byte (0x10) is not part of the private tag, it just references the private block where it is stored.

Sorry I had missed this the first time...
